<launch>
  <include file="$(find realsense_camera)/launch/r200_nodelet_rgbd.launch">
    <arg name="depth_width" value="640"/>
    <arg name="depth_height" value="480"/>
    <arg name="rgb" default="color"/>
  </include>
  <node name="cone_finder" pkg="robo_magellan" type="detect_cones.py">
    <rosparam command="load" file="$(find robo_magellan)/launch/cone_finder.yaml" />
    <param name="~captureVideo" value="True" />
    <param name="~publishImages" value="True" />
    <param name="~ignorePriorDetections" value="True" />

    <!-- Use a very small cone area since we area filtering
         in the navigator. -->
    <param name="~minConeArea" value="75" />

    <!-- Reject cone detections below this aspect ratio. -->
    <param name="~min_aspect_ratio" value="1.1" />

    <!-- Reject cone contours with more than this number of vertices. -->
    <param name="~max_contour_vertices" value="1000000" />

    <!-- Require cones to be below the horizon. -->
    <param name="~max_y" value="10000" />

    <!-- Use 'bin' for new binning algorithm, 'hsv' for hue-based algorithm. -->
    <param name="~thresholdAlgorithm" value="bin" />
    <param name="~binConfig" value="$(find robo_magellan)/config/realsense-bins-0_6.json" />

    <!-- Use 'convexHull', 'relaxed' or 'huMoments' for filter algorithm. -->
    <param name="~contourFilterAlgorithm" value="relaxed" />
  </node>
</launch>

